### PR TITLE
New `In` predicate to test for manifest membership

### LINKED ISSUE #49: A few new functions proposed to impement for `type Manifest struct`
```
type Manifest struct {
	resources []unstructured.Unstructured
	Client    Client
	log       logr.Logger
}
```

I propose to have a few functions implemented for this Manifest type.
1. 
```
func (m Manifest) RemoveResources(resources []unstructured.Unstructured) error {
	...
	return nil
}
```
This function will remove all the resources as listed in `resources []unstructured.Unstructured`. After we compare two Manifest, we can generate a list of resources to be delete from the cluster. I think this function should be handy.

2.
```
func (m Manifest) SetResources(resources []unstructured.Unstructured) error {
	...
	return nil
}
```
This function will set all the resources as listed in `resources []unstructured.Unstructured`.  It will overwrite all the existing resources.

3.
```
func (m Manifest) AddResources(resources []unstructured.Unstructured) error {
	...
	return nil
}
```
This function adds the resources into the existing resources in Manifest.

### PR DESCRIPTION
**Pull request by jcrossley3**
Fixes #49

### CODE REVIEW

File: filter.go [hunk 1]
```diff
@@ -107,3 +110,18 @@ func ByGVK(gvk schema.GroupVersionKind) Predicate {
 		return u.GroupVersionKind() == gvk
 	}
 }
+
+// In returns true if present in the passed Manifest, using
+// "gvk|namespace/name" as a unique identifier
+func In(manifest Manifest) Predicate {
```
Comment (markusthoemmes) [inline filter.go:116]: This feels odd, shouldn't this take a GVK, namespace and name? The description says that this checks if a resource is in a manifest. This reads like it checks that the entire manifest is in another manifest. Am I missing something?

Comment (jcrossley3) [reply filter.go:116]: Well, it's returning a `Predicate`, which is intended to filter *another* manifest. How would you suggest making it less odd?

File: filter_test.go [hunk 2]
```diff
@@ -169,3 +169,12 @@ func TestConvertFilter(t *testing.T) {
...
 	}
 }
+
+func TestInFilter(t *testing.T) {
+	eleven, _ := NewManifest("testdata/k-s-v0.11.0.yaml")
+	twelve, _ := NewManifest("testdata/k-s-v0.12.1.yaml")
+	new := twelve.Filter(None(In(eleven)))
+	if len(new.Resources()) != 1 {
+		t.Error("Missing the autoscaler-hpa")
+	}
```
Comment (houshengbo) [inline filter_test.go:179]: This PR will address the issue I have. 
Do I understand correctly?
`twelve.Filter(None(In(eleven)))` means locating the resources exist in `twelve`, but not in `eleven`.
`twelve.Filter(In(eleven))` means locating the resources exist both in `twelve` and `eleven`.

File: filter_test.go [hunk 3]
```diff
@@ -169,3 +169,12 @@ func TestConvertFilter(t *testing.T) {
 		}
 	}
 }
+
+func TestInFilter(t *testing.T) {
+	eleven, _ := NewManifest("testdata/k-s-v0.11.0.yaml")
+	twelve, _ := NewManifest("testdata/k-s-v0.12.1.yaml")
+	new := twelve.Filter(None(In(eleven)))
+	if len(new.Resources()) != 1 {
+		t.Error("Missing the autoscaler-hpa")
```
Comment (houshengbo) [inline filter_test.go:178]: Should be missing the Service autoscaler-hpa.

File: filter_test.go (hunk 2 above)
Comment (jcrossley3) [reply filter_test.go:179]: Yes, exactly. It's essentially the *intersection* of the two. Makes me wonder whether `Intersect` is a better name than `In`, though I like the conciseness of this when finding which resources are now obsolete:
```go
  deletes := v1.Filter(Not(In(v2)))
```

File: filter.go (hunk 1 above)
Comment (markusthoemmes) [reply filter.go:116]: I read it wrong, it's fine I think.

File: filter_test.go (hunk 2 above)
Comment (houshengbo) [reply filter_test.go:179]: Cool, we got it.

**Pull request merged**